sopel/irc/backends.py

```diff
@@ -72,38 +96,39 @@ class AsynchatBackend(AbstractIRCBackend, asynchat.async_chat):
     :param bot: a Sopel instance
     :type bot: :class:`sopel.bot.Sopel`
     :param int server_timeout: connection timeout in seconds
-    :param int ping_timeout: ping timeout in seconds
+    :param int ping_interval: ping interval in seconds
     """
-    def __init__(self, bot, server_timeout=None, ping_timeout=None, **kwargs):
+    def __init__(self, bot, server_timeout=None, ping_interval=None, **kwargs):
         AbstractIRCBackend.__init__(self, bot)
         asynchat.async_chat.__init__(self)
         self.writing_lock = threading.RLock()
         self.set_terminator(b'\n')
         self.buffer = ''
         self.server_timeout = server_timeout or 120
-        self.ping_timeout = ping_timeout or (self.server_timeout / 2)
+        self.ping_interval = ping_interval or (self.server_timeout * 0.45)
```

Comment:
Aside from needing to tweak the docs for the config option _again_, any reason not to do this?

```suggestion
        self.ping_interval = ping_interval or int(self.server_timeout * 0.45)
```

I'll note that using `0.4` instead of `0.45` would mean the ping interval comes out to a whole number whenever the timeout is a multiple of 5, which I would bet represents _almost all_ timeout values used in the wild… but now I'm just nerding out over the internal number that no one will ever see. 😅

Reply:
A float is fine by me in this case, we don't need the accuracy of an int, and we can definitively allow a rounding error.